js/webapp.js

```diff
@@ -884,8 +889,37 @@ WebApp.prototype = {
 
   installErrorHanders() {
     this.expressApp.use((req, res, next) => {
-      do404(req.url, res, this.options.productCode
-          + ": unknown resource requested");
+      const headers = req.headers
+      for (const header of Object.keys(headers)) {
+        /* Try to find a referer header and try to
+         * redirect to our server,
+         */
+        if (header == 'referer') {
+          var pattern = /^http.+\/ZLUX\/plugins\/.+/;
+          if (pattern.test(headers[header])) {
+            var parts = headers[header].split("/");
+            var zluxIndex = parts.indexOf("ZLUX");
+            var pluginID = parts[zluxIndex + 2];
+            var serviceName = parts[zluxIndex + 4];
+            var myProxy = proxyMap.get(pluginID + ":" + serviceName);
+            var fullUrl = req.originalUrl;
```

Comment:
This part of the code has one area where it falls short. Basically, if I went to http://localhost:12360/ZLUX/plugins/com.rs.zosmf.proxy/services/data instead of http://localhost:12360/ZLUX/plugins/com.rs.zosmf.proxy/services/data**/** then the path lookup for HTML would omit the **data** part of the URL. 

It would be good to try and detect relative pathing to a minor degree to see that if the requested URL appears to be some subject of the referrer URL, then likely this is a relative path gone awry.

This is not really a bug for the PR in that this behavior existed regardless, so I'll still merge.

![image](https://user-images.githubusercontent.com/30730276/46979078-9a7b9280-d09e-11e8-9a75-291b0e78310a.png)